### PR TITLE
Additional commands for ventilator control

### PR DESCRIPTION
**Pull request by janvkem**

### CODE REVIEW

File: ramses_rf/protocol/command.py [hunk 1]
```diff
@@ -389,13 +391,13 @@ def set_fan_rate(
         #  I 218 --:------ --:------ 39:159057 22F1 003 000204  # low
 
         # Type 2: 21:038634 18:126620 --:------ (less common)
-        #  - are cast as a triplet, 0.085s apart, without a seqn (---)
+        #  - are cast as a triplet, 0.085s apart, with a no seqn (---)
```
Comment (Kars-de-Jong) [inline ramses_rf/protocol/command.py]: That doesn't seem right... "with a no seqn".

**Comment by Kars-de-Jong** [comment]
Great that you're adding this!
How generic are these commands? Will they work for both Itho and Orcon systems?
They seem to use the same opcodes, but not the same payloads...

File: ramses_rf/protocol/command.py (hunk 1 above)
Comment (zxdavb) [reply ramses_rf/protocol/command.py]: You are not looking at the latest code. @janvkem needs to rebase.
```python
        # Type 2: 21:038634 18:126620 --:------ (less common)
        #  - are cast as a triplet, 0.085s apart, without a seqn (---)
        #  - only payloads seen: '000.0[47A]'
        #  I --- 21:038634 18:126620 --:------ 22F1 003 000507
```

**Comment by zxdavb** [comment]
> Great that you're adding this! How generic are these commands? Will they work for both Itho and Orcon systems? They seem to use the same opcodes, but not the same payloads...

Yes, I am not convinced I fully understand the differences - I am waiting for more packet logs from people.

I think I will open up with very basic functionality (the one without the sequence number) & that will drive people to contribute.

**Comment by Kars-de-Jong** [comment]
> Yes, I am not convinced I fully understand the differences - I am waiting for more packet logs from people.

Sorry, can't contribute there yet, I ordered an SSM-D2 yesterday, and I get the keys for my new house next month.

File: ramses_rf/protocol/command.py (hunk 1 above)
Comment (janvkem) [reply ramses_rf/protocol/command.py]: Yes sorry, that is my mistake. Still new to using github and have to find a good workflow. I'm now committing by hand via the web interface which does not work that well. I will pay a bit more attention to my next PRs

**Comment by janvkem** [comment]
> Great that you're adding this! How generic are these commands? Will they work for both Itho and Orcon systems? They seem to use the same opcodes, but not the same payloads...

Which commands do you mean? I added quite a few. I believe the 2411 are specific Orcon. 22F1 is same for orcon and itho except that they have a different mapping, that is why the servicecall allows 0 1 2 3 inputs instead of away low medium high. 22F3 I don't know, but most itho commands are known I believe, so it could be compared. In case they are different it might be an option to add vender as additional input and change commands based on that. I only have Orcon to look at :)

**Comment by janvkem** [comment]
> > Yes, I am not convinced I fully understand the differences - I am waiting for more packet logs from people.
> 
> Sorry, can't contribute there yet, I ordered an SSM-D2 yesterday, and I get the keys for my new house next month.

Lots of itho commands can be found here:
https://github.com/arjenhiemstra/ithowifi/blob/6dee42d18c1c3192aeca6dd27dfbd0e75a2e5778/software/NRG_itho_wifi/IthoPacket.h

**Comment by janvkem** [comment]
@zxdavb I have no idea how to fix the unsuccesful checks that are left. My python knowledge is lacking here. Can you take a look and suggest changes?

Edit: Is it as simple as removing the @property?

**Comment by zxdavb** [comment]
@janvkem Your PR is too 'big' - please start with just one command: something like **`set_bypass_position`**, aka `W|22F7`.

You will need, in the one PR:
 - payload schema in **ramses.py**
 - parser in **parser.py**
 - command constructor in **command.py**
 - constants in **const.py**

and, _importantly_:
 - test in **tests/test_put_set_apis.py** - but create a new one called **test_apis_hvac.py**

Have a look at how it is done for (say) `W|2309`.

There are several reasons to start with just the one command, trust me - it will be better for both of us.

**Comment by zxdavb** [comment]
BTW, there should only be one command constructor per verb|code pair - for example, I see you've multiple such for `W|2411`.

**Comment by zxdavb** [comment]
OK, I got bored and (almost) finished `W|22F7` - have a look - submit a PR for each constructor... one issue is that your PR is causing me to make changes to my branch, so will be easier in small chunks

**Comment by zxdavb** [comment]
You haven't included a parser for `2411` - ~I have started one - please complete the lookup table~ (in English, I am sorry - I will look at translating one day).
```python
    LOOKUP_2411 = {
        "3D": "Away mode Supply fan rate (%)",
        "3E": "Away mode Exhaust fan rate (%)",
    }
```
Of course, I am willing to have the above translations corrected,

edit: I have finished the table now

**Comment by janvkem** [comment]
I will start splitting up the big PR into smaller chunks and take a look at the 2411 parser.

**Comment by zxdavb** [comment]
@janvkem 

Please don't take offence - your Python skills mean you might struggle to submit a PR I could merge without a lot of work at your end...

You could contribute more effectively if you collected annotated packet logs, pressed buttons on your system, etc. The packet log you contributed fro `2411` was very useful...

Are you geared up for doing things like this:
```
python client.py monitor -x "W 37:171871 32:155617 2411 00003D000F00000018" -o packet.log
```

(there is a chance this would work)

If so, I can tweak **client.py** for you.

The idea is that you can validate protocol-level features I want to add to **ramses_rf** without us having to go to the expense of fleshing out the corresponding code in **ramses_cc**, just so you can test them.

Let me know what you think.

**Comment by janvkem** [comment]
> @janvkem
> 
> Please don't take offence - your Python skills mean you might struggle to submit a PR I could merge without a lot of work at your end...
> 
> You could contribute more effectively if you collected annotated packet logs, pressed buttons on your system, etc. The packet log you contributed fro `2411` was very useful...
> 
> Are you geared up for doing things like this:
> 
> ```
> python client.py monitor -x "W 37:171871 32:155617 2411 00003D000F00000018" -o packet.log
> ```
> 
> (there is a chance this would work)
> 
> If so, I can tweak **client.py** for you.
> 
> The idea is that you can validate protocol-level features I want to add to **ramses_rf** without us having to go to the expense of fleshing out the corresponding code in **ramses_cc**, just so you can test them.
> 
> Let me know what you think.

I'm not easily offended. I am fully aware that my python skills are lacking. To be honest I was already wondering if I was not annoying you with my pull requests with lots of errors :)

Running these commands is something I can easily do. Output in packet.log for the one above (with a `29:xxxxxx` instead of a `37:xxxxxx`:

```text
2022-06-19T19:48:07.352510 000  W --- 29:158183 32:155617 --:------ 2411 009 00003D000F00000018
2022-06-19T19:48:07.392521 057  I --- 32:155617 29:158183 --:------ 2411 023 00003D370F00000018000000000000005000000001A632
```
and output on screen:
```text
19:48:07.352  W --- 29:158183 32:155617 --:------ 2411 009 00003D000F00000018 < Corrupt payload: Payload doesn't match '^0000[0-9A-F]{42}$': 00003D000F00000018
19:48:07.392 || HVC:155617 |  29:158183 |  I | fan_params       |      || {'parameter': '3D', 'description': 'Away mode Supply fan rate (%)', 'value': 0.12, 'min_value': 0.0, 'max_value': 0.4, 'precision': 0.005, '_value_06': '370F', '_value_42': 'A632'}
```
I took a 29 device instead of a 37 as that is paired to my remote and I already verified before that also works. If you want to see it with a 37 I can also pair a 37 remote.

**Comment by zxdavb** [comment]
Perfect - so we've learnt something useful: the HRU will respond to an abbreviated `2411` command (I thought it might). The next step would be to confirm that the requested change was effected.

We should move from this platform to gitter, at: https://gitter.im/evohome_rf/community

BTW, there is no difference between a `37:` and a `29:` (this is true fro HVAC, but not strictly true fro HEAT) - all that really matters is that you have the correct **destination_id**, and that device is willing to accept commands from the **source_id**.

You can also try a 'new' **device_id** `29:123456` to confirm we do need to bind first.

**Comment by janvkem** [comment]
> Perfect - so we've learnt something useful: the HRU will respond to an abbreviated `2411` command (I thought it might). The next step would be to confirm that the requested change was effected.
> 
> We should move from this platform to gitter, at: https://gitter.im/evohome_rf/community
> 
> BTW, there is no difference between a `37:` and a `29:` (this is true fro HVAC, but not strictly true fro HEAT) - all that really matters is that you have the correct **destination_id**, and that device is willing to accept commands from the **source_id**.
> 
> You can also try a 'new' **device_id** `29:123456` to confirm we do need to bind first.

Replied on gitter

**Pull request closed without merging**